src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_named_user.erl

```diff
@@ -178,12 +178,59 @@ from_json(Req, #base_state{resource_args = invitation_response,
                     {true, chef_wm_util:set_json_body(Req1, EJResponse), State1}
             end
     end;
-from_json(Req, #base_state{server_api_version = ?API_v0,
-                           resource_state = #user_state{ chef_user = User, user_data = UserData}} = State) ->
-    oc_chef_wm_key_base:update_object_embedded_key_data_v0(Req, State, User, UserData);
-from_json(Req, #base_state{resource_state = #user_state{ chef_user = User, user_data = UserData}} = State) ->
+from_json(Req, #base_state{server_api_version = ApiVersion, resource_state =
+    #user_state{ chef_user = #chef_user{email = OrigEmail0} = User, user_data = UserData0}} = State) ->
+    % After this change all the email addresses stored in lowercase
+    % If any email already exist with case sensitive, We transform it into lower for comparison
+    % And treat them unique. if a new user trying to use the same email with different case
+    % then rejecting it
+    UserData =
+    case ej:get({<<"email">>}, UserData0) of
+        undefined ->
+            UserData0;
+        Email ->
+            ej:set({<<"email">>}, UserData0, string:lowercase(Email))
+    end,
+    OrigEmail =
+    case OrigEmail0 of
+        undefined -> OrigEmail0;
+        null -> undefined;
+        _ -> string:lowercase(OrigEmail0)
+    end,
+    NewEmail = ej:get({<<"email">>}, UserData),
+    EmailUpdateConfig = envy:get(oc_chef_wm, allow_email_update_only_from_manage, boolean),
+    %% check if the request originated from the webui or client (client /knife)
+    RequestOrigin = wrq:get_req_header("x-ops-request-source", Req),
     % in v1+ keys may only be updated via the keys endpoint.
-    oc_chef_wm_base:update_from_json(Req, State, User, UserData).
+
+    % Input arguments for case
+    % NewEmail - From Update record
+    % RequestOrigin - Request originated from webui or clinet
+    % EmailUpdateConfig - allow_email_update_only_from_manage config value
+    % ApiVersion - API version
+
+    % Different case clauses
+    % case 1 &  2- Orginal Email is same or undefined then Update
+    % case 3 & 4 - Request from Chef Manage/ Web then then Update
+    % case 5 & 6 - Request is not from Chef Manage/Web & allow_email_update_only_from_manage false then Update
+    % Other cases - Reject with 403
+    case {NewEmail, RequestOrigin, EmailUpdateConfig, ApiVersion} of
+        {OEmail, _, _, ?API_v0} when OEmail =:= OrigEmail; OEmail =:= undefined ->
+            oc_chef_wm_key_base:update_object_embedded_key_data_v0(Req, State, User, UserData);
+        {OEmail, _, _, _} when OEmail =:= OrigEmail; OEmail =:= undefined ->
+            oc_chef_wm_base:update_from_json(Req, State, User, UserData);
+        {_, "web", _, ?API_v0} ->
+            oc_chef_wm_key_base:update_object_embedded_key_data_v0(Req, State, User, UserData);
+        {_, "web", _, _} ->
+            oc_chef_wm_base:update_from_json(Req, State, User, UserData);
+        {_, Origin, false, ?API_v0} when Origin =/= "web" ->
```

Comment:
We can remove the guard here - Origin can't be `web` because that would be handled in the case above  `{_, "web", _, ?API_v0}` or the next `{_, "web", _, _}.

Reply:
sure - I see that adding the guard makes it more readable, but might not be necessary for the code.